.github/workflows/release.yml

```diff
@@ -3,42 +3,73 @@ on:
   push:
     branches:
       - master
+  # TODO: uncomment before merging
+  pull_request:
 
 jobs:
-  stable:
-    uses: the-guild-org/shared-config/.github/workflows/release-stable.yml@main
-    with:
-      releaseScript: release
-      nodeVersion: 18
-      packageManagerVersion: modern
-    secrets:
-      githubToken: ${{ secrets.GITHUB_TOKEN }}
-      npmToken: ${{ secrets.NODE_AUTH_TOKEN }}
-
-  docker-image:
-    needs: stable
-    if: needs.stable.outputs.published == 'true'
+  # TODO: uncomment before merging
+  # stable:
+  #   uses: the-guild-org/shared-config/.github/workflows/release-stable.yml@main
+  #   with:
+  #     releaseScript: release
+  #     nodeVersion: 18
+  #     packageManagerVersion: modern
+  #   secrets:
+  #     githubToken: ${{ secrets.GITHUB_TOKEN }}
+  #     npmToken: ${{ secrets.NODE_AUTH_TOKEN }}
+
+  serve-with-hive-docker-image:
+    # needs: stable
+    # TODO: uncomment before merging
+    # if: needs.stable.outputs.published == 'true'
     runs-on: ubuntu-latest
+
     steps:
       - name: checkout
         uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
 
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v3
       - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v3
-      - name: Login to DockerHub
-        uses: docker/login-action@v3
+        uses: docker/setup-buildx-action@v2
+
+      - name: Create Docker Buildx builder
+        run: docker buildx create --name mybuilder --use
+
+      - name: Inspect Docker Buildx builder
+        run: docker buildx inspect --bootstrap
+
+      - name: Log in to DockerHub
+        uses: docker/login-action@v2
         with:
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_PASSWORD }}
-      - name: Build & Publish Docker Image
-        id: docker_build
+
+      - name: Build Docker image for local testing
+        id: build-single
         uses: docker/build-push-action@v6
         with:
+          context: ./docker
+          push: false
+          load: true
+          platforms: linux/amd64
+          tags: theguild/graphql-mesh:test
+
+      - name: Run E2E tests
+        run: |
+          cd ./docker
+          chmod +x e2e-docker-test.sh
+          ./e2e-docker-test.sh
+
+      - name: Build and push multi-platform Docker image
+        if: success()
+        id: build-multi
+        uses: docker/build-push-action@v6
+        with:
+          context: ./docker
           push: true
+          platforms: linux/amd64,linux/arm64
           tags: |
-            theguild/graphql-mesh:latest
+            theguild/graphql-mesh:v1
             theguild/graphql-mesh:${{ github.sha }}
-          context: ./packages/container
-          file: ./packages/container/Dockerfile
```

Comment:
Let's publish to the GitHub registry instead for the reasons we discussed before.

Here some example from yoga and hive

Hive (bake file)
https://github.com/kamilkisiela/graphql-hive/blob/bd736899633aeaf838cb09b152f35fb86c1f3dc0/.github/workflows/build-and-dockerize.yaml#L127-L147

Yoga (docker-compose)
https://github.com/dotansimha/graphql-yoga/pull/3329/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR298-R309